fullrt/dht.go

```diff
@@ -831,40 +831,58 @@ func (dht *FullRT) Provide(ctx context.Context, key cid.Cid, brdcst bool) (err e
 }
 
 func (dht *FullRT) execOnMany(ctx context.Context, fn func(context.Context, peer.ID) error, peers []peer.ID) int {
+	if len(peers) == 0 {
+		return 0
+	}
+
 	putctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	waitAllCh := make(chan struct{}, len(peers))
+	waitFailCh := make(chan struct{})
+	waitSuccessCh := make(chan struct{})
```

Comment:
I'm pretty sure the lack of buffering here is really hurting performance, I'm going to increase the buffers to `len(peers)` and see how we're doing.

Reply:
add the buffering helps, but it's still much slower than previously (previously it took about 15s to send out 1% of 1M records and now it's about 5 min).

Going to do more poking around

Reply:
The only reason that buffering should be any help is because the channel reader is busy doing something other than reading the channel.  This should no longer be the case with the fix to stop resetting the timer on every iteration.  I would be very surprised/interested if a buffer of more than 1 made any difference.